registry/manifest_test.go

```diff
@@ -0,0 +1,120 @@
+package registry
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/opencontainers/go-digest"
+)
+
+var (
+	fakeDigest = digest.FromString("sha256:0000000000000000000000000000000000000000000000000000000000000000")
+)
+
+func NewHandlerFunc(mediaType string) func(w http.ResponseWriter, r *http.Request) {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if strings.Contains(r.URL.Path, "emptyheader") {
+			w.Header().Add("Docker-Content-Digest", "")
+		}
+
+		if strings.Contains(r.URL.Path, "invalidheader") {
+			w.Header().Add("Docker-Content-Digest", "invaliddigest")
+		}
+
+		if strings.Contains(r.URL.Path, "validheader") {
+			w.Header().Add("Docker-Content-Digest", string(fakeDigest))
+		}
+
+		io.WriteString(w, fmt.Sprintf(`{"mediaType":"%s"}`, mediaType))
```

Comment:
nit: I think this is equivalent to `w.Write([]byte(fmt.Sprintf(`{"mediaType":"%s"}`, mediaType)))` which means we'd have one fewer import if you want to switch to that

Reply:
https://github.com/stackrox/docker-registry-client/pull/20/commits/fd1ec934915c2a83f3103495ee227563ff7ef8c0